### PR TITLE
[SD-2906],[SD-2907] package publish validations

### PR DESCRIPTION
**Pull request by marwoodandrew**

### CODE REVIEW

**Comment by amagdas** [comment]
:+1:

File: server/apps/publish/archive_publish.py [hunk 1]
```diff
@@ -595,6 +583,34 @@ def _update_headline_sequence(self, doc):
...
+    def _validate_package_contents(self, package, takes_package):
+        """
+        If the item passed is a package this function will ensure that the unpublished content validates and none of
+         the content is locked by other than the publishing session, also do not allow any killed or spiked content
+        :param package:
+        :param takes_package:
+        :raises: Validation exceptions if the validation fails
+        """
+        # Ensure it is the sort of thing we need to validate
+        if package[ITEM_TYPE] == CONTENT_TYPE.COMPOSITE and not takes_package and self.publish_type == 'publish':
```
Comment (plamut) [inline server/apps/publish/archive_publish.py]: I think we have a constant defined for publish types (created recently in the other pull request)?

**Comment by mugurrus** [comment]
:+1:

**Comment by plamut** [comment]
:+1:

(see the minor remark if it applies)

File: server/apps/publish/archive_publish.py (hunk 1 above)
Comment (marwoodandrew) [inline server/apps/publish/archive_publish.py]: Ah yes, I looked but not well enough!

**Pull request merged**